org.adempiere.base/src/org/compiere/model/MOrder.java

```diff
@@ -1406,49 +1406,41 @@ protected boolean afterSave (boolean newRecord, boolean success)
 	{
 		if (!success || newRecord)
 			return success;
-		
-		// TODO: The changes here with UPDATE are not being saved on change log - audit problem  
-		
-		//	Propagate Description changes
-		if (is_ValueChanged("Description") || is_ValueChanged("POReference"))
-		{
-			String sql = "UPDATE C_Invoice i"
-				+ " SET (Description,POReference)="
-					+ "(SELECT Description,POReference "
-					+ "FROM C_Order o WHERE i.C_Order_ID=o.C_Order_ID) "
-				+ "WHERE DocStatus NOT IN ('RE','CL') AND C_Order_ID=" + getC_Order_ID();
-			int no = DB.executeUpdateEx(sql, get_TrxName());
-			if (log.isLoggable(Level.FINE)) log.fine("Description -> #" + no);
-		}
-
-		//	Propagate Changes of Payment Info to existing (not reversed/closed) invoices
-		if (is_ValueChanged("PaymentRule") || is_ValueChanged("C_PaymentTerm_ID")
-			|| is_ValueChanged("C_Payment_ID")
-			|| is_ValueChanged("C_CashLine_ID"))
-		{
-			String sql = "UPDATE C_Invoice i "
-				+ "SET (PaymentRule,C_PaymentTerm_ID,C_Payment_ID,C_CashLine_ID)="
-					+ "(SELECT PaymentRule,C_PaymentTerm_ID,C_Payment_ID,C_CashLine_ID "
-					+ "FROM C_Order o WHERE i.C_Order_ID=o.C_Order_ID)"
-				+ "WHERE DocStatus NOT IN ('RE','CL') AND C_Order_ID=" + getC_Order_ID();
-			//	Don't touch Closed/Reversed entries
-			int no = DB.executeUpdate(sql, get_TrxName());
-			if (log.isLoggable(Level.FINE)) log.fine("Payment -> #" + no);
-		}
-	      
-		//	Propagate Changes of Date Account to existing (not completed/reversed/closed) invoices
-		if (is_ValueChanged("DateAcct"))
-		{
-			String sql = "UPDATE C_Invoice i "
-				+ "SET (DateAcct)="
-					+ "(SELECT DateAcct "
-					+ "FROM C_Order o WHERE i.C_Order_ID=o.C_Order_ID)"
-				+ "WHERE DocStatus NOT IN ('CO','RE','CL') AND Processed='N' AND C_Order_ID=" + getC_Order_ID();
-			//	Don't touch Completed/Closed/Reversed entries
-			int no = DB.executeUpdate(sql, get_TrxName());
-			if (log.isLoggable(Level.FINE)) log.fine("DateAcct -> #" + no);
-		}
-	      
+
+		// Propagate changes to not-completed/reversed/closed invoices
+		String[] propagateCols = new String[] {
+				COLUMNNAME_Description,	
+				COLUMNNAME_POReference,
+				COLUMNNAME_PaymentRule,
+				COLUMNNAME_C_PaymentTerm_ID,
+				COLUMNNAME_DateAcct};
+		boolean propagateColChanged = false;
```

Comment:
what would be the options available if an implementation wish to change the auto propagate columns ?

Reply:
more columns can be propagated with an EventHandler

less columns or a different document status would not be possible, maybe an EventHandler on MInvoice to capture the change and reset it back to the old value?

any suggestion?

Reply:
perhaps as a new sysconfig entry ?

Reply:
> perhaps as a new sysconfig entry ?

ready, changed the approach to use SysConfig ORDER_COLUMNS_TO_COPY_TO_NOT_COMPLETED_INVOICES